packages/rs-drive/src/drive/identity/withdrawals/documents.rs

```diff
@@ -97,56 +98,61 @@ impl Drive {
         Ok(outcome.documents_owned())
     }
 
-    /// Find one document by it's transactionId field
-    pub fn find_withdrawal_document_by_transaction_id(
+    // TODO(withdrawals): Currently it queries only up to 100 documents.
+    //  It works while we don't have pooling
+
+    /// Find up to 100 documents by status and transaction indices
+    pub fn find_up_to_100_withdrawal_documents_by_status_and_transaction_indices(
         &self,
-        original_transaction_id: &[u8],
+        status: withdrawals_contract::WithdrawalStatus,
+        transaction_indices: &[WithdrawalTransactionIndex],
         transaction: TransactionArg,
         platform_version: &PlatformVersion,
-    ) -> Result<Document, Error> {
-        let data_contract_id = withdrawals_contract::ID;
-
-        let contract_fetch_info = self
-            .get_contract_with_fetch_info_and_fee(
-                data_contract_id.to_buffer(),
-                None,
-                true,
-                transaction,
-                platform_version,
-            )?
-            .1
-            .ok_or_else(|| {
-                Error::Drive(DriveError::CorruptedCodeExecution(
-                    "Can't fetch data contract",
-                ))
-            })?;
-
-        let document_type = contract_fetch_info
-            .contract
-            .document_type_for_name(withdrawal::NAME)?;
-
+    ) -> Result<Vec<Document>, Error> {
```

Comment:
maybe put the limit as a parameter? Also this should be versioned.

Reply:
Same question https://github.com/dashpay/platform/pull/1601#discussion_r1479198183